### PR TITLE
Added graphs for custom field arguments per date

### PR DESCRIPTION
**Pull request by purmonen**

### CODE REVIEW

File: recruitment/views.py [hunk 1]
```diff
@@ -205,7 +210,8 @@ def json(self):
                 'monocolor': self.monocolor,
                 'charts': self.charts,
                 'sorted_values': self.sorted_values(),
-                'sorted_value_counts': self.sorted_value_counts()
+                'sorted_value_counts': self.sorted_value_counts(),
+                'y_limit': self.y_limit
```
Comment (mictab) [inline recruitment/views.py]: Add a comma at the end of this line in case someone wants to add an item in the future.

File: recruitment/views.py [hunk 2]
```diff
@@ -246,6 +252,40 @@ def json(self):
...
+                for argument in arguments:
+                    date_dictionary = dict(
+                        [(date_filter(application.submission_date, "d M"), application.submission_date) for application in
+                         application_list])
+
+                    argument_per_date_count = ValueCounter(
+                        argument.value,
+                        True,
+                        ['bar'],
+                        [date_filter(application.submission_date, "d M") for application in application_list if CustomFieldAnswer.objects.filter(user=application.user, answer=str(argument.pk)).exists()],
```
Comment (mictab) [inline recruitment/views.py]: Too long line, violates PEP8.

File: recruitment/views.py [hunk 3]
```diff
@@ -246,6 +252,40 @@ def json(self):
...
+                        lambda x: date_dictionary[x]
```
Comment (mictab) [inline recruitment/views.py]: Add comma at the end of the line.

File: recruitment/views.py [hunk 4]
```diff
@@ -246,6 +252,40 @@ def json(self):
...
+                    )
+
+                    for date in daterange(recruitment_period.start_date, min(timezone.now(), recruitment_period.end_date)):
```
Comment (mictab) [inline recruitment/views.py]: Too long line.

File: recruitment/views.py [hunk 5]
```diff
@@ -246,6 +252,40 @@ def json(self):
...
+                        date_string = date_filter(date, "d M")
+                        if not date_string in argument_per_date_count.data:
```
Comment (mictab) [inline recruitment/views.py]: Please change to `if date_string not in argument_per_date_count.data:` as this complies with PEP8.

File: recruitment/views.py [hunk 6]
```diff
@@ -180,10 +180,15 @@ def __init__(self, description, monocolor, charts, values=None, sort_key=None):
             self.monocolor = monocolor
             self.charts = charts
             self.sort_key = sort_key
+            self.y_limit = None
 
             if values:
                 self.add_values(values)
 
+
+        def set_y_limit(self, y_limit):
```
Comment (mictab) [inline recruitment/views.py]: If you really need to use a setter (the instance variables are public anyway) please rather use the `@property` decorator together with `y_limit`.

File: recruitment/views.py [hunk 7]
```diff
@@ -180,10 +180,15 @@ def __init__(self, description, monocolor, charts, values=None, sort_key=None):
             self.monocolor = monocolor
             self.charts = charts
             self.sort_key = sort_key
+            self.y_limit = None
 
             if values:
                 self.add_values(values)
 
+
```
Comment (mictab) [inline recruitment/views.py]: Remove this blank line.

**Comment by macaullyjames** [comment]
As discussed in Slack, this should be no longer be merged into `test` so there's no need to change the target branch 👍

**Comment by purmonen** [comment]
So now I have updated this branch. I guess that automatically means this pull request would merge my new changes? If so plz review again :)

**Pull request merged**